File: backend/eBooking/Persistence/Models/RoomMaintenanceLog.cs
namespace Persistence.Models
{
    public class RoomMaintenanceLog : BaseEntity
    {
        public int RoomId { get; set; }
        public DateTime ReportedAt { get; set; }
        public DateTime? ResolvedAt { get; set; }
        public string Description { get; set; } = string.Empty;
        public decimal Cost { get; set; }
        public string TechnicianName { get; set; } = string.Empty;
        public Room Room { get; set; } = null!;
    }
}

File: Desktop/hotel_system_desktop/lib/models/room_maintenance_log.dart
class RoomMaintenanceLog {
  final int id;
  final int roomId;
  final String roomNumber;
  final DateTime reportedAt;
  final DateTime? resolvedAt;
  final String description;
  final double cost;
  final String technicianName;

  RoomMaintenanceLog({
    required this.id,
    required this.roomId,
    required this.reportedAt,
    this.resolvedAt,
    required this.description,
    required this.cost,
    required this.technicianName,
    this.roomNumber = '',
  });

  String get roomDisplayLabel =>
      roomNumber.isNotEmpty ? roomNumber : 'Soba #$roomId';

  factory RoomMaintenanceLog.fromJson(Map<String, dynamic> json) =>
      RoomMaintenanceLog(
        id: json['id'] ?? 0,
        roomId: json['roomId'] ?? 0,
        roomNumber: json['roomNumber'] ?? '',
        reportedAt:
            DateTime.tryParse(json['reportedAt'] ?? '') ?? DateTime.now(),
        resolvedAt: json['resolvedAt'] != null
            ? DateTime.tryParse(json['resolvedAt'])
            : null,
        description: json['description'] ?? '',
        cost: (json['cost'] ?? 0).toDouble(),
        technicianName: json['technicianName'] ?? '',
      );
}
